### DISCUSSION TITLE
Tui Input Date Validation

### DISCUSSION THREAD

**Discussion by Joshua-Ryan** [taiga-family/taiga-ui#1750] [Q&A] [answered]
So while using the input date I have come to realize that it does not emit partial values. This is great for not saving incorrect values, however it makes it hard to notify the user if a field is inputted correctly. This is because validation runs off of emitted values, so I cannot check if it is a partial value if it only emits full values. 

Has anyone run into this issue, and if so what did you do to notify the user that certain fields are not full?

**Comment by waterplea** [comment, marked as answer]
There's a special directive validator for this case, works for all Taiga controls that behave the same way (time, date range etc). See first example:
https://taiga-ui.dev/components/input-date#base
